src/rpc/rawtransaction.cpp

```diff
@@ -196,7 +196,7 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
             }
             errmsg = "No such transaction found in the provided block";
         } else if (!g_txindex) {
-            errmsg = "No such mempool transaction. Use -txindex to enable blockchain transaction queries";
+            errmsg = "No such mempool transaction. Use -txindex or provide a block hash to enable blockchain transaction queries";
```

Comment:
No strong opinion, but generally I think error messages shouldn't mention arguments or configurations, for that we have the help message.

Reply:
Is that really a guideline? This seems very helpful, especially as the underlying API changes(no more UTXO lookup)

Reply:
For cli usage this might be useful, but for the developer what really matters IMO is the documentation and error code in the response.

I understand the issue regarding the API change, but I'd rather see a documentation improvement than having "hints" in the errors.